### PR TITLE
Pass broker class to reconciler tests

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Pass broker class to reconciler tests
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
/lgtm

**Comment by pierDipi** [comment]
```
/tmp/go-build3730747384/b001/e2e_new.test flag redefined: cluster
panic: /tmp/go-build3730747384/b001/e2e_new.test flag redefined: cluster
goroutine 1 [running]:
flag.(*FlagSet).Var(0xc000114180, {0x32bc188, 0xc0006a24c0}, {0x2c726af, 0x7}, {0x2cb26f2, 0x2e})
	/root/.gvm/gos/go1.18.3/src/flag/flag.go:879 +0x3a5
flag.(*FlagSet).StringVar(...)
	/root/.gvm/gos/go1.18.3/src/flag/flag.go:762
knative.dev/pkg/environment.(*ClientConfig).InitFlags(0xc0006a24c0, 0x8?)
	/home/prow/go/src/knative.dev/eventing-kafka-broker/vendor/knative.dev/pkg/environment/client_config.go:40 +0x96
knative.dev/pkg/injection.ParseAndGetRESTConfigOrDie()
	/home/prow/go/src/knative.dev/eventing-kafka-broker/vendor/knative.dev/pkg/injection/config.go:32 +0x65
knative.dev/pkg/injection.EnableInjectionOrDie({0x0, 0x0}, 0x0)
	/home/prow/go/src/knative.dev/eventing-kafka-broker/vendor/knative.dev/pkg/injection/injection.go:46 +0xe5
knative.dev/eventing-kafka-broker/test/e2e_new.TestMain(0xc000100568?)
	/home/prow/go/src/knative.dev/eventing-kafka-broker/test/e2e_new/main_test.go:68 +0xb2
main.main()
	_testmain.go:65 +0x318
FAIL	knative.dev/eventing-kafka-broker/test/e2e_new	0.114s
```

File: test/e2e_new/main_test.go [hunk 1]
```diff
@@ -36,6 +37,7 @@ import (
 	"knative.dev/reconciler-test/pkg/environment"
 
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/kafka"
+	brokertest "knative.dev/eventing-kafka-broker/test/pkg/broker"
```
Comment (pierDipi) [inline test/e2e_new/main_test.go:40]: This additional import might be causing the above error because of the init function's side effects.

Comment (aliok) [reply test/e2e_new/main_test.go]: Thanks for catching that. This is the flaky test effect: I overlooked that there's actually a problem!

File: test/reconciler-tests.sh [hunk 2]
```diff
@@ -27,6 +27,8 @@ if [ "${EVENTING_KAFKA_BROKER_CHANNEL_AUTH_SCENARIO:-""}" != "" ]; then
   success
 fi
 
+export BROKER_CLASS="Kafka"
```
Comment (matzew) [inline test/reconciler-tests.sh:30]: will we run them eventually w/ `NamespacedBrokerClass`  / `KafkaNamespaced` too?

**Review by pierDipi — APPROVED** [review]
/lgtm

File: test/reconciler-tests.sh (hunk 2 above)
Comment (aliok) [reply test/reconciler-tests.sh:30]: Yes, I will. I will do that in a separate PR.

That actually needs some prow job changes too

**Pull request merged**